### PR TITLE
CA-236821: Provide clear error message if GPG key is not imported.

### PR DESCRIPTION
**Pull request by huizh**
If the update required gpg key is not imported in the host,
the precheck will fail. But the error message is unclear.
It says unknown error.

To fix it, add error code UPDATE_PRECHECK_FAILED_GPGKEY_NOT_IMPORTED
and return clear error message.

Signed-off-by: Hui Zhang <hui.zhang@citrix.com>

### CODE REVIEW

File: ocaml/idl/datamodel.ml [hunk 1]
```diff
@@ -1216,6 +1216,8 @@ let _ =
     ~doc:"The update precheck stage failed: the server is of an incorrect version." ();
   error Api_errors.update_precheck_failed_out_of_space ["update"; "available_space"; "required_space "]
     ~doc:"The update precheck stage failed: the server does not have enough space." ();
+  error Api_errors.update_precheck_failed_gpgkey_not_imported ["update"]
+    ~doc:"The update precheck stage failed: the update required gpg key is not imported in the host." ();
```
Comment (thomassa) [inline ocaml/idl/datamodel.ml]: We use a GPG signature for validating the hotfix as a whole, and then the `yum` command checks GPG signatures on individual RPM packages.
This error corresponds to an error message from `yum`.
Therefore please would you make a change?
`the update required gpg key is not imported in the host.`
would be better as something like
`RPM package validation requires a GPG key that has not been imported on the host."

Comment (huizh) [inline ocaml/idl/datamodel.ml]: @thomassa, thanks for the suggestion. That makes sense.
Update the message accordingly.
Could you please take a look?

Thanks.

File: ocaml/idl/datamodel.ml [hunk 2]
```diff
@@ -1216,6 +1216,8 @@ let _ =
     ~doc:"The update precheck stage failed: the server is of an incorrect version." ();
   error Api_errors.update_precheck_failed_out_of_space ["update"; "available_space"; "required_space "]
     ~doc:"The update precheck stage failed: the server does not have enough space." ();
+  error Api_errors.update_precheck_failed_gpgkey_not_imported ["update"]
+    ~doc:"The update precheck stage failed: RPM package validation requires a GPG key that is not imported on the host." ();
```
Comment (lindig) [inline ocaml/idl/datamodel.ml]: Just a nit pick for native speakers to review: is a _key that is not imported on this host_ the right tense? I would have expected _key has not been imported into this host_ or _key is not present on this host_.

Comment (huizh) [inline ocaml/idl/datamodel.ml]: HI @lindig and @thomassa, I updated the message to _key is not present on the host_. Is that OK?

Comment (thomassa) [inline ocaml/idl/datamodel.ml]: I would say any of those would be reasonable.

"...*is* not imported..." is like saying some wood *is* not varnished or a machine *is* not switched on or a light-bulb *is* not installed in a fitting.

In principle there could be a key which *has been* imported but then removed.

Talking about the key being present avoids the question over tense, but it a bit less specific about what kind of presence is meant, and is a bit further from the language used in the original error message from the gpg tool.

**Comment by thomassa** [comment]
This looks good to me now. Merging.

**Pull request merged**